config/web/middleware-dispatcher.php

```diff
@@ -5,14 +5,16 @@
 use Psr\Container\ContainerInterface;
 use Yiisoft\Csrf\CsrfMiddleware;
 use Yiisoft\ErrorHandler\ErrorCatcher;
+use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 use Yiisoft\Router\Middleware\Router;
 use Yiisoft\Session\SessionMiddleware;
-use Yiisoft\Yii\Web\MiddlewareDispatcher;
 
 return [
-    MiddlewareDispatcher::class => static fn (ContainerInterface $container) => (new MiddlewareDispatcher($container))
-        ->addMiddleware($container->get(Router::class))
-        ->addMiddleware($container->get(SessionMiddleware::class))
-        ->addMiddleware($container->get(CsrfMiddleware::class))
-        ->addMiddleware($container->get(ErrorCatcher::class)),
+    MiddlewareDispatcher::class => static fn (ContainerInterface $container) => $container->get(MiddlewareDispatcher::class)
```

Comment:
New instance should be created because likely the bug is because it's shared w/ router.